app/services/network_topology_service.rb

```diff
@@ -0,0 +1,117 @@
+class NetworkTopologyService < TopologyService
+  def initialize(provider_id)
+    @provider_id = provider_id
+    @providers = retrieve_providers(ManageIQ::Providers::NetworkManager, @provider_id)
+    # @entity = retrieve_entity(entity_id, entity_type)
+    @entity = @providers.first
+  end
+
+  def entity_type(entity)
+    if entity.kind_of?(CloudNetwork)
+      entity.class.base_class.name.demodulize
+    else
+      entity.class.name.demodulize
+    end
+  end
+
+  def build_topology
+    topo_items = {}
+    links = []
+
+    entity_relationships = {
+      :NetworkManager => {
+        :CloudSubnets => {
+          :CloudNetwork => nil,
+          :Vms => {
+            :FloatingIps => nil,
+            :CloudTenant => nil,
+            :SecurityGroups => nil
+          },
+          :NetworkRouter => {
+            :CloudNetwork => nil
+          },
+        }
+      },
+    }
+    # loaded = @providers.includes(:cloud_subnets => [:cloud_network,
```

Comment:
> btw. to see what queries your code does...

console should already be in debug and showing the queries...If not, it's easier to just set the level or use our helper method:

```
ActiveRecord::Base.logger.level = 1

# or

enable_console_sql_logging
```